src/Foundation/NSOperatingSystemVersion.cs

```diff
@@ -36,20 +36,102 @@ namespace Foundation  {
 	[SupportedOSPlatform ("macos")]
 	[SupportedOSPlatform ("tvos")]
 #endif
-	// NSProcessInfo.h
-	public struct NSOperatingSystemVersion {
+
+	public struct NSOperatingSystemVersion: IEquatable<NSOperatingSystemVersion>, IComparable{
 		public nint Major, Minor, PatchVersion;
 		
-		public NSOperatingSystemVersion (nint major, nint minor, nint patchVersion)
+        public NSOperatingSystemVersion (nint majorVersion, nint minorVersion, nint patchVersionArg)
+		{
+			Major = majorVersion;
+			Minor = minorVersion;
+			PatchVersion = patchVersionArg;
+		}
+
+        public NSOperatingSystemVersion (nint majorVersion, nint minorVersion)
 		{
-			Major = major;
-			Minor = minor;
-			PatchVersion = patchVersion;
+			Major = majorVersion;
+			Minor = minorVersion;
+			PatchVersion = 0;
 		}
 
+        public NSOperatingSystemVersion (nint majorVersion)
+		{
+			Major = majorVersion;
+			Minor = 0;
+			PatchVersion = 0;
+		}
+
+        public int CompareTo (NSOperatingSystemVersion otherVersion) {
+            int majorValue = Major.CompareTo(otherVersion.Major);
+            if (majorValue == 0) {
+                int minorValue = Minor.CompareTo(otherVersion.Minor);
+                if (minorValue == 0) {
+                    int patchValue = PatchVersion.CompareTo(otherVersion.PatchVersion);
+                    return patchValue;
+                }
+                else {
+                    return minorValue;
+                } 
+            }
+            else {
+                return majorValue;
+            }
+                
+        }
+
+        public int CompareTo(Object obj) {
+            if (obj == null) return 1;
+            if (obj is NSOperatingSystemVersion other)
+                return CompareTo(other);
+            return 1;
+        }
+
 		public override string ToString ()
 		{
 			return Major + "." + Minor + "." + PatchVersion;
 		}
+
+		public bool Equals (NSOperatingSystemVersion other) 
+        {
+            if (other == null)
```

Comment:
Lets add at the top of the file:
```cs
#nullable enable
```
That will force you to make a few changes. On would be to do:
```suggestion
            if (other is null)
```